src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -164,8 +164,16 @@ public void actionPerformed(ActionEvent e) {
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
         if (s != null && s.length() > 0) {
-          clientContext.user.addUser(s);
-          UserPanel.this.getAllUsers(listModel);
+          //if alphanumeric
+          if(s.matches("[a-zA-Z0-9]+")) {
+            clientContext.user.addUser(s);
+            UserPanel.this.getAllUsers(listModel);
+          } else {
+            JOptionPane.showMessageDialog(UserPanel.this,
+                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
+                "User Not Created",
+                JOptionPane.ERROR_MESSAGE);
+          }
```

Comment:
I think all the code below is actually repetitive.
ClientCpnext.user.addUser(s):
1.user is a ClientUser object and addUser(username) in that class will be called.
2.inside addUser(), there is isValidName() called, so whether the name is valid is checked there.
The errors have also been handled along the way.
whether the name is valid is checked in ClientUser class
```
 +          if(s.matches("[a-zA-Z0-9]+")) {
 +            clientContext.user.addUser(s);
 +            UserPanel.this.getAllUsers(listModel);
 +          } else {
 +            JOptionPane.showMessageDialog(UserPanel.this,
 +                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
 +                "User Not Created",
 +                JOptionPane.ERROR_MESSAGE);
 +          }

```